File: ParadigmaComp/15Questoes/golang/questao9.go
package main

// Interfaces/Protocolos

import "fmt"

// Interface Imprimivel
type Imprimivel interface {
	Imprimir() string
}

// Estrutura Relatorio
type Relatorio struct{}

func (r Relatorio) Imprimir() string {
	return "Imprimindo Relatório..."
}

// Estrutura Contrato
type Contrato struct{}

func (c Contrato) Imprimir() string {
	return "Imprimindo Contrato..."
}

func executarImpressao(imprimivel Imprimivel) {
	fmt.Println(imprimivel.Imprimir())
}

func main() {
	relatorio := Relatorio{}
	contrato := Contrato{}

	executarImpressao(relatorio)
	executarImpressao(contrato)
}

File: ParadigmaComp/15Questoes/python/questao9.py
from abc import ABC, abstractmethod

# Interfaces/Protocolos

class Imprimivel(ABC):
    @abstractmethod
    def imprimir(self):
        pass

class Relatorio(Imprimivel):
    def imprimir(self):
        return "Imprimindo Relatório..."

class Contrato(Imprimivel):
    def imprimir(self):
        return "Imprimindo Contrato..."

# Exemplo de uso
def executar_impressao(documento: Imprimivel):
    print(documento.imprimir())

relatorio = Relatorio()
contrato = Contrato()

executar_impressao(relatorio)
executar_impressao(contrato)
